map/framework.cpp

```diff
@@ -1650,11 +1650,52 @@ void Framework::ConnectToGpsTracker()
 void Framework::DisconnectFromGpsTracker()
 {
   m_connectToGpsTrack = false;
-  GpsTracker::Instance().Disconnect();
+  auto & tracker = GpsTracker::Instance();
+  tracker.Disconnect();
+  tracker.SetEnabled(false);
   if (m_drapeEngine)
     m_drapeEngine->ClearGpsTrackPoints();
 }
 
+void Framework::StartTrackRecording()
+{
+  auto & tracker = GpsTracker::Instance();
+  if (!tracker.IsEnabled())
+    tracker.SetEnabled(true);
+  m_connectToGpsTrack = true;
+  if (m_drapeEngine)
+  {
+    m_drapeEngine->ClearGpsTrackPoints();
+    tracker.Connect(bind(&Framework::OnUpdateGpsTrackPointsCallback, this, _1, _2));
+  }
+}
+
+void Framework::StopTrackRecording()
+{
+  m_connectToGpsTrack = false;
+  auto & tracker = GpsTracker::Instance();
+  tracker.Disconnect();
+  tracker.SetEnabled(false);
+  if (m_drapeEngine)
+    m_drapeEngine->ClearGpsTrackPoints();
```

Comment:
This should be done only if the track was saved, right?

Reply:
Yes!

Reply:
As a result of the discussion I leave the 
```
if (m_drapeEngine)
    m_drapeEngine->ClearGpsTrackPoints();
```
only in the `SaveRecordedTrackWithName`. So the dotted line will not be removed from the screen when `recent track` (for the android) or track recording (for the ios) is stopped without saving.